[KOGITO-589] bash script should have bash shebang

gabriel376 opened the pull request:
[KOGITO-859](https://issues.jboss.org/browse/KOGITO-589)

Script ```hack/go-build-cli``` has bash-specific features such as function keyword and arrays. If shebang is #!/bin/sh and sh is a link to any shell different from bash (dash, for example), the script will fail. Making an explicit shebang #!/bin/bash resolves the issue.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [X] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [X] Pull Request contains link to the JIRA issue
- [X] Pull Request contains description of the issue
- [X] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

spolti commented:
Why exactly did you change it? have you find an issue?

gabriel376 commented:
I changed the shebang of hack/go-build-cli.sh from #!/bin/sh to #!/bin/bash.
This script is not POSIX compliant and has bash-specific features. I am running Debian Buster, which has sh linked to dash and this script fails to run.

spolti commented:
thanks for the explanation, if it is really necessary, please do:
- open a ticket for it under kogito project.
- instead use /bin/bash use **#!/usr/bin/env bash** for portability purposes.

I think this one will work well on all *nix systems.

gabriel376 commented:
Created https://issues.jboss.org/browse/KOGITO-589

gabriel376 commented:
Changes of commit 3ee1aa93379859abdf246fc024880df5efc84849:
* modified shebang to **#!/bin/env bash** of all hack scripts that are not POSIX compliant
* replaced legacy backtick \`pwd\` with **$(pwd)** notation 
* used **double quote** on shell variables to prevent globbing

ricardozanini commented:
@spolti are we ready?

spolti commented:
@ricardozanini  yes
@gabriel376 thanks!
